### DISCUSSION TITLE
For each Server, select a Snippet to "auto run" when connecting

### DISCUSSION THREAD

**Discussion by ehlersd** [isontheline/pro.webssh.net#639] [Ideas]
This would be great for positioning the CWD (for example), using multiple "Servers" that connect to the same Host, but with a different set of Snippets and functionality.

**Comment by isontheline** [comment]
Could you explain more about this feature?

Does it's about this? https://github.com/isontheline/pro.webssh.net/issues/564

**Reply by ehlersd** [reply]
Not really tied to that other issue.

Previously, I was using another SSH Mac App that had been deprecated, so it was time to update.  Your app was the closest to what I needed.  Specifically, I manage numerous Firebase Projects, updating functions, hosting images, uploading Crashlytics symbols from CI builds, etc.  

So, for example, I have a "ME-Functions" Server for my ME project.  It just connects to localhost, but then I have to do a CD call:

`cd ~/development/me/functions `

which I also follow by a Firebase command:

`firebase projects:list`

to confirm whether I'm currently on the DEV or PROD project.  

I have these 2 commands in a "StartUp" snippet that I have to run each time I connect.  It would be a whole lot easier if I could specify that the ME-Functions Server when it connects could automatically run the "StartUp" snippet (or whichever snipped I specified for that System.

**Reply by isontheline** [reply]
Maybe you can use the "execute command" feature :
<img width="1425" alt="image" src="https://user-images.githubusercontent.com/44212923/171486807-1e8a26c1-3bb4-4cb5-a6d7-84a4220dce76.png">

It will allow you to run a command on connection. 
It's not related to snippets though.

You may try : 

`cd ~/development/me/functions; firebase projects:list; bash -l`

**Reply by ehlersd** [reply]
Okay, I had tried this previously, but I don't use the shell a whole lot and forgot the bash -l at the end.  Makes sense now (other than using zsh, but same thing).  I managed to get it to do what I needed it to.

BUT, I will add, the edit popup for the Custom execute command is a single-line, short edit field.....a challenge to use.  So, I would still love to have it run a snippet after fully connecting....but also can live without it.

**Reply by isontheline** [reply]
No problem I'm opening an issue to track implementation of your request : https://github.com/isontheline/pro.webssh.net/issues/640
